ports/stm32f4/supervisor/usb.c

```diff
@@ -79,7 +81,21 @@ void init_usb_hardware(void) {
     HAL_GPIO_Init(GPIOG, &GPIO_InitStruct);
     never_reset_pin_number(0, 8);
 #endif
+    
+#ifdef BOARD_NO_VBUS
+    disable_usb_vbus();
+#endif
```

Comment:
Shouldn't this go after the peripheral is turned on below?

Reply:
*blinks vacantly*

yes. 
yes, it should

Reply:
@hierophect you need to explicitly enable VBUS for board that doesn't define `BOARD_NO_VBUS` as well. Since that code is removed from dcd_synopsys. As Scott mention, this must be after the `__HAL_RCC_USB_OTG_FS_CLK_ENABLE()`

```
  // Enable VBUS hardware sensing, enable pullup, enable peripheral.
#ifdef USB_OTG_GCCFG_VBDEN
  USB_OTG_FS->GCCFG |= USB_OTG_GCCFG_VBDEN;
#else
  USB_OTG_FS->GCCFG |= USB_OTG_GCCFG_VBUSBSEN;
#endif
```